trk/src/Screens/GymDaily/index.js

```diff
@@ -1,8 +1,219 @@
+  const [totalClimbs, setTotalClimbs] = useState('');
+  const [asents, setAscents] = useState([]);
+
+  const [totalAscents, setTotalAscents] = useState('');
+  const [latestAscents, setLatestAscents] = useState('');
+  const [mostCompleted, setMostCompleted] = useState('');
+  const [leastCompleted, setLeastCompleted] = useState('');
+  const [peakTime, setPeakTime] = useState('');
+  const [highestRated, setHighestRated] = useState({
+    name: '',
+    grade: '',
+    type: '',
+    climb: '',
+  });
+  const [latestFeedback, setLatestFeedback] = useState({
+    explanation: '',
+    climb: '',
+    rating: '',
+    climbName: '',
+  });
+
+  const { currentUser } = useContext(AuthContext);
+  const { getClimbsBySomeField } = ClimbsApi();
+  const { getTapsBySomeField } = TapsApi();
+  const { getCommentsBySomeField } = CommentsApi();
+  const { getClimb } = ClimbsApi();
+
+  const userId = currentUser.uid;
+
+
+  useEffect(() => {
+    const fetchClimbs = async () => {
+      try {
+        const querySnapshot = await getClimbsBySomeField('setter', userId);
+        if (querySnapshot && querySnapshot.docs) {
+          const climbs = querySnapshot.docs.map(doc => {
+            return {
+              id: doc.id,
+              ...doc.data()
+            };
+          }).filter(climb => !climb.archived); // Filter out archived climbs
+
+          setYourClimbs(climbs);
+          setTotalClimbs(climbs.length.toString()); // Now reflects the count of non-archived climbs
+        } else {
+          console.error("No data found in Firestore query");
+        }
+      } catch (error) {
+        console.error("Error fetching climbs:", error);
+      }
+    };
+
+    fetchClimbs();
+    console.log(`Your climbs are: ${JSON.stringify(yourClimbs, null, 2)}`);
+
+  }, [userId]);
+
+
+  useEffect(() => {
+    const fetchTapsAndCalculateAscents = async () => {
+      let totalTaps = 0;
+      let allTaps = [];
+      let latestAscentsCount = 0;
+      const oneDayAgo = new Date(Date.now() - 24 * 60 * 60 * 1000);
+      let climbFrequencyMap = {};
+      let timeFrequencyMap = {};
+
+
+
+      for (const climb of yourClimbs) {
+        const tapsSnapshot = await getTapsBySomeField('climb', climb.id);
+        if (tapsSnapshot && tapsSnapshot.docs) {
+          const taps = tapsSnapshot.docs.map(doc => {
+            const tap = doc.data();
+            tap.timestamp = tap.timestamp.toDate();
+            console.log(`Tap Timestamp: ${tap.timestamp}`);
+            return tap;
+          });
+          totalTaps += taps.length;
+          allTaps = [...allTaps, ...taps];
+
+          latestAscentsCount += taps.filter(tap => tap.timestamp > oneDayAgo).length;
+
+          taps.forEach(tap => {
+            if (!climbFrequencyMap[tap.climb]) {
+              climbFrequencyMap[tap.climb] = { count: 0, name: climb.name, id: climb.id, grade: climb.grade };
+            }
+            climbFrequencyMap[tap.climb].count++;
+          });
+        }
+      }
+
```

Comment:
using the list of climbs you have set (which has previously been fetched above), we fetch all taps associated to these climbs and count the total taps. We also count the total taps in the last 24hrs using the timestamp value of the taps. we also calculate the number of taps associated with each climb in your set (so that we can later calculate the most completed route)